propagation/b3/spancontext.go

```diff
@@ -0,0 +1,76 @@
+package b3
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/openzipkin/zipkin-go/model"
+)
+
+func parseHeaders(
+	traceIDHeader, spanIDHeader, parentSpanIDHeader, sampledHeader, flagsHeader string,
+) (*model.SpanContext, error) {
+	var (
+		err           error
+		spanID        uint64
+		requiredCount int
+		sc            = &model.SpanContext{}
+	)
+
+	switch strings.ToLower(sampledHeader) {
+	case "0", "false":
+		sampled := false
+		sc.Sampled = &sampled
+	case "1", "true":
```

Comment:
good.. maybe add comment about leniency for old clients

Reply:
will do